### PR TITLE
Mqtt extension

### PR DESCRIPTION
**Pull request by hjw2015**
added a more detailed MQTT publishing function to enable visualizing via e.g. home automation systems (like OpenHab) easily.

### CODE REVIEW

File: sunray/comm.cpp [hunk 1]
```diff
@@ -1053,6 +1056,12 @@ void mqttCallback(char* topic, byte* payload, unsigned int length) {
   }
 }
 
+// define a macro so avoid repetitive code lines for sending single values via MQTT
+#define MQTT_SEND(VALUE, FORMAT, TOPIC) \
```
Comment (jkuettner) [inline sunray/comm.cpp]: Perhaps a little more meaningful in the mqtt context:

```suggestion
#define MQTT_PUBLISH(VALUE, FORMAT, TOPIC) \
```

**Comment by disaster123** [comment]
@hjw2015 can you please clean up this branch? There should be no merge commits in it also no reverts or cleanups. Please squash or fixup this and rebase it.

File: sunray/comm.cpp [hunk 2]
```diff
@@ -1061,13 +1070,61 @@ void processWifiMqttClient()
...
+      MQTT_SEND(statMowBumperCounter, "%d", "/stats/mow/bumperEvents")
+      MQTT_SEND(statMowSonarCounter, "%d", "/stats/mow/sonarEvents")
+      MQTT_SEND(statMowLiftCounter, "%d", "/stats/mow/liftEvents")
+      MQTT_SEND(statMowMaxDgpsAge, "%.2f", "/stats/mow/maxDgpsAge")
+      MQTT_SEND(statMowDistanceTraveled, "%.1f", "/stats/mow/distanceTraveled")
+      MQTT_SEND(statMowInvalidRecoveries, "%d", "/stats/mow/invalidRecoveries")
+      MQTT_SEND(statImuRecoveries, "%d", "/stats/imuRecoveries")
+      MQTT_SEND(statGPSJumps, "%d", "/stats/gpsJumps")
+      MQTT_SEND(statTempMin, "%.1f", "/stats/tempMin")
+      MQTT_SEND(statTempMax, "%.1f", "/stats/tempMax")
```
Comment (jkuettner) [inline sunray/comm.cpp]: is it possibly also useful to include the current temperatures?

**Comment by hjw2015** [comment]
Closed in favor of #107

**Pull request closed without merging**